src/type/definition.ts

```diff
@@ -319,6 +360,7 @@ export function assertAbstractType(type: unknown): GraphQLAbstractType {
  * ```
  */
 export class GraphQLList<T extends GraphQLType> {
+  readonly [isGraphQLListTypeSymbol]?: true = true;
```

Comment:
TypeScript went wild without the optional, anyone having an idea why?

Reply:
Hmmm. What is the error message?

Reply:
<img width="919" alt="image" src="https://user-images.githubusercontent.com/14338007/171166514-119f371e-c56a-475f-9399-650bf9e6a611.png">

```
Type 'GraphQLType' does not satisfy the constraint 'GraphQLNullableType'.ts(2344)
```

at https://github.com/n1ru4l/graphql-js/blob/dc18ec7c0c5ba0c89a5e074102b784ba57c1dc53/src/type/definition.ts#L441

Reply:
<img width="965" alt="image" src="https://user-images.githubusercontent.com/14338007/171167308-ed8b6b21-941a-4cc9-93af-ee773f75b354.png">

```
Type 'GraphQLType' does not satisfy the constraint 'GraphQLNullableType'.
  Type 'GraphQLNonNull<GraphQLNullableType>' is not assignable to type 'GraphQLNullableType'.
    Property '[isGraphQLListTypeSymbol]' is missing in type 'GraphQLNonNull<GraphQLNullableType>' but required in type 'GraphQLList<GraphQLType>'.ts(2344)
```

at https://github.com/n1ru4l/graphql-js/blob/dc18ec7c0c5ba0c89a5e074102b784ba57c1dc53/src/type/definition.ts#L227

Reply:
I think that's a separate error in definition of `GraphQLWrappingType` somehow uncovered only by this PR. I have submitted a separate fix #3622.